File: 0240_Search a 2D Matrix II/Searcha2DMatrixII.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m = matrix.length;
        if(m == 0) return false;
        int n = matrix[0].length;
        
        int r = 0;
        int c = n-1;
        
        while(r < m && c >= 0)
        {
            if(matrix[r][c] == target) return true;
            else if(matrix[r][c] > target)
                c--;
            else
                r++;
        }
        
        return false;
    }
}

File: 0240_Search a 2D Matrix II/Searcha2DMatrixII.cs
public class Solution {
    public bool SearchMatrix(int[][] matrix, int target) {
        var m = matrix.Length;
        if(m == 0) return false;
        var n = matrix[0].Length;
        
        int row = 0;
        int col = n-1;
        
        while(row < m && col >= 0)
        {
            if(matrix[row][col] == target) return true;
            else if(matrix[row][col] > target)
                col--;
            else
                row++;
        }
        
        return false;
    }
}